torch/utils/data/standard_pipes.ipynb

```diff
@@ -0,0 +1,1046 @@
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(40).batch(2).batch(4).batch(3).unbatch(unbatch_level = -1)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "## Map\n",
+    "\n",
+    "Function: `map`\n",
+    "\n",
+    "Description: \n",
+    "\n",
+    "Alternatives:\n",
+    "\n",
+    "Arguments:\n",
+    "  - `nesting_level: int = 0`\n",
+    " \n",
+    "Example:"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "0\n2\n4\n6\n8\n10\n12\n14\n16\n18\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10).map(lambda x: x * 2)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "`map` by default applies function to every mini-batch as a whole\n"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[0, 1, 2, 0, 1, 2]\n[3, 4, 5, 3, 4, 5]\n[6, 7, 8, 6, 7, 8]\n[9, 9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10).batch(3).map(lambda x: x * 2)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "To apply function on individual items of the mini-batch use `nesting_level` argument"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[[0, 2, 4], [6, 8, 10]]\n[[12, 14, 16], [18]]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10).batch(3).batch(2).map(lambda x: x * 2, nesting_level = 2)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "Setting `nesting_level` to `-1` will apply `map` function to the lowest level possible"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[[[0, 2, 4], [6, 8, 10]], [[12, 14, 16], [18]]]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10).batch(3).batch(2).batch(2).map(lambda x: x * 2, nesting_level = -1)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "## Filter\n",
+    "\n",
+    "Function: `filter`\n",
+    "\n",
+    "Description: \n",
+    "\n",
+    "Alternatives:\n",
+    "\n",
+    "Arguments:\n",
+    "  - `nesting_level: int = 0`\n",
+    "  - `drop_empty_batches = True` whether empty many batches dropped or not.\n",
+    " \n",
```

Comment:
- `nesting_level` is 1
It's pretty straight forward in this case
```py
dp
[[1, 2], 3]
[[4, 5], 6]
[[7, 8], 9]

dp.filter(lambda x: x[0] < 5 if isinstance(x, list) else x > 5, nesting_level=1)
[[1, 2]]
[[4, 5], 6]
[9]
```
- `nesting_level` is 2 or -1
If element has lower level than `nesting_level`, how should we handle it? 
Example: `dp.filter(lambda x: x > 4, nesting_level=2)`
  - Apply to filter function with the lowest level element
```py
[[5], 6]
[[7, 8], 9]
```
  - Drop the element that can not reach the `nesting_level`
```py
[[5]]
[[7, 8]]
```